src/bucket/BucketManager.h

```diff
@@ -205,10 +210,13 @@ class BucketManager : NonMovableOrCopyable
     virtual std::vector<std::string>
     checkForMissingBucketsFiles(HistoryArchiveState const& has) = 0;
 
-    // Restart from a saved state: find and attach all buckets in `has`, set
-    // current BL. Pass `maxProtocolVersion` to any restarted merges.
-    virtual void assumeState(HistoryArchiveState const& has,
-                             uint32_t maxProtocolVersion) = 0;
+    // Assume state from `has` in BucketList: find and attach all buckets in
+    // `has`, set current BL. Note: Does not restart merging
+    virtual void assumeState(HistoryArchiveState const& has) = 0;
```

Comment:
Hmm, not sure I understand why merge restart is now done outside of assumeState. I think separating functionality like this (set state, build index, start merges) is generally a wrong pattern: what we should be doing is centralizing functionality to restore bucket state in one place, as it's less error-prone and allows us to reason about what's supposed to happen at state restoration (previously, this was assumeState). What you can do is turn assumeState into a work that executes a sequence of events (we have WorkSequence for this): set state, rebuild indexes (this must be done in the background as per my other comment, using IndexBucketWork you already implemented), restart merges. Then in catchup you can add it to the sequence after bucket application. On startup, you'd just call executeWork for synchronous execution.